### ISSUE TITLE
Blue and white colours of class - setting one for both

### ISSUE THREAD

**Issue by theb4ku** [dracula/visual-studio-code#171]
Hey, can you guys help me to navigate which line I should change in .json file to change white colour to blue? As in the screenshot, the **Program** class is blue but the **Thread** class is white and I would like to change that to also blue.
![image](https://user-images.githubusercontent.com/75274385/111907531-117a9980-8a56-11eb-9df5-1a9860c33c80.png)

**Comment by dsifford** [comment]
@theb4ku do you have a minimal example file I can take a look at? Difficult to say otherwise.

It looks like you have semantic highlighting enabled for that given the hover info, but unsure.

**Comment by theb4ku** [comment]
@dsifford [https://github.com/theb4ku/dracula-Issue/blob/main/Program.cs](url)
Check this out, and for example in 20-22 lines class are blue but in line 36 class is white. I am sure this occurs when I use classes and their methods from other namespaces for example **class.Method();** And it's same problem with parameters in lambda or methods - it's possible to set parameter's color to orange too in body method & in lamda?
![image](https://user-images.githubusercontent.com/75274385/112212521-63b9e700-8c1d-11eb-8346-66339c7c7e9f.png)
![image](https://user-images.githubusercontent.com/75274385/112212704-98c63980-8c1d-11eb-89ca-cf6b357eff42.png)

**Comment by dsifford** [comment]
@theb4ku Ah, thank you for the example.

The issue is due to the syntax engine of vscode. For anything other than (IIRC) JavaScript and TypeScript, vscode uses TextMate syntaxes which applies colors with the use of complex regular expressions.

Because of the fact that these regexes can't be aware of context, it's unable to know that the use of classes are actually classes and not like variables or something else.

So in the context of `Task`, on line 20 and 21, the first use is colored blue because the regex is able to deduce that it's a type. The second use on those lines, the regex is unable to determine what it is, so it sets it as a variable.

Hope that makes sense.

In short, the only way to fix that is to use a better syntax engine (e.g. [tree-sitter](https://github.com/tree-sitter/tree-sitter))

**Comment by theb4ku** [comment]
Hey, I figure it out how to fix it. In settings we need to change `Editor › Semantic Highlighting: Enabled` to `true` and for c# language we need to check box in `Csharp › Semantic Highlighting: Enabled`. Thanks for answers @dsifford because of that I checked this two options.
